test/opt/test_set_spec_const_default_value.cpp

```diff
@@ -20,6 +20,120 @@ using namespace spvtools;
+        // 9. minimum spec id
+        {"{0:100}", true, SpecIdToValueStrMap{{0, "100"}}},
+        // 10. random content without spaces are allowed
+        {"{200: random_stuff}", true,
+         SpecIdToValueStrMap{{200, "random_stuff"}}},
+        // 11. missing ending '}' is fine for parsing
+        {"{100:1024", true, SpecIdToValueStrMap{{100, "1024"}}},
+        // 12. multiple ending '}' is fine for parsing
+        {"{100:1024}}", true, SpecIdToValueStrMap{{100, "1024"}}},
+        // 13. support spec id in hex format (just because we use the
+        // ParseNumber() utility)
+        {"{0x100:1024}", true, SpecIdToValueStrMap{{256, "1024"}}},
+        // 14. tailing comma and spaces is fine
+        {"{100:1024, 200:2048, \r \n \t  }", true,
+         SpecIdToValueStrMap{{100, "1024"}, {200, "2048"}}},
+        // 15. multiple entries
+        {"{101:1, 102:2, 103:3, 104:4, 200:201, 9999:1000, 0x100:333}", true,
+         SpecIdToValueStrMap{{101, "1"},
+                             {102, "2"},
+                             {103, "3"},
+                             {104, "4"},
+                             {200, "201"},
+                             {9999, "1000"},
+                             {256, "333"}}},
+        // 16. ignore the content after '}'
+        {"{100:1024, 200:2048,} something irrelevant", true,
+         SpecIdToValueStrMap{{100, "1024"}, {200, "2048"}}},
+    }));
+
+INSTANTIATE_TEST_CASE_P(
+    InvalidString, DefaultValuesStringParsingTest,
+    ::testing::ValuesIn(std::vector<DefaultValuesStringParsingTestCase>{
+        // 0. only spec id
+        {"{100:}", false, SpecIdToValueStrMap{}},
+        // 1. spec id is not an integer
+        {"{100.0:}", false, SpecIdToValueStrMap{}},
+        // 2. spec id is not a number
+        {"{something_not_a_number:1}", false, SpecIdToValueStrMap{}},
+        // 3. space within spec id
+        {"{100 0:20}", false, SpecIdToValueStrMap{}},
+        // 4. same spec id defined multiple times
+        {"{100:20, 100:21}", false, SpecIdToValueStrMap{}},
+        // 5. missing spec id
+        {"{:3}", false, SpecIdToValueStrMap{}},
+        // 6. missing beginning '{'
+        {"100:1024}", false, SpecIdToValueStrMap{}},
+        // 7. multiple beginning '{'
+        {"{{100:1024}", false, SpecIdToValueStrMap{}},
+        // 8. only colon
+        {"{:}", false, SpecIdToValueStrMap{}},
+        // 9. spec id overflow
+        {"{4294967296:0}", false, SpecIdToValueStrMap{}},
+        // 10. spec id less than 0
+        {"{-1:0}", false, SpecIdToValueStrMap{}},
+        // 11. empty string
+        {"", false, SpecIdToValueStrMap{}},
+        // 12. empty comma is not allow
+        {
+            "{100:1024,,200:2048}", false, SpecIdToValueStrMap{},
+        },
+        // 13. empty comma is not allow
+        {"{,}", false, SpecIdToValueStrMap{}},
+        // 14. missing comma
+        {"{100:1024 200:2048}", false, SpecIdToValueStrMap{}},
```

Comment:
Hmm, I thought some of them are already covered by current test cases. Anyway they will be added.